subsys/net/l2/canbus/canbus_raw.c

```diff
@@ -7,22 +7,36 @@
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(net_l2_canbus, LOG_LEVEL_NONE);
 
+#include <zephyr/drivers/can.h>
 #include <zephyr/net/net_core.h>
 #include <zephyr/net/net_l2.h>
 #include <zephyr/net/net_if.h>
 #include <zephyr/net/net_pkt.h>
 #include <zephyr/net/canbus.h>
+#include <zephyr/net/ethernet.h>
 
-static inline enum net_verdict canbus_recv(struct net_if *iface,
-					   struct net_pkt *pkt)
+static inline enum net_verdict canbus_recv(struct net_if *iface, struct net_pkt *pkt)
 {
+	struct can_frame frame;
+
 	memset(net_pkt_lladdr_src(pkt)->addr, 0, sizeof(net_pkt_lladdr_src(pkt)->addr));
 	net_pkt_lladdr_src(pkt)->len = 0U;
 	net_pkt_lladdr_src(pkt)->type = NET_LINK_CANBUS_RAW;
 	memset(net_pkt_lladdr_dst(pkt)->addr, 0, sizeof(net_pkt_lladdr_dst(pkt)->addr));
 	net_pkt_lladdr_dst(pkt)->len = 0U;
 	net_pkt_lladdr_dst(pkt)->type = NET_LINK_CANBUS_RAW;
 
+	if (net_pkt_get_len(pkt) < sizeof(struct can_frame)) {
+		return NET_DROP;
+	}
+
+	memcpy(&frame, net_pkt_data(pkt), sizeof(struct can_frame));
```

Comment:
Copying the data to the frame variable seems pointless, we have all the data already in pkt.
You could create the "frame" variable using the `NET_PKT_DATA_ACCESS_DEFINE()` macro which uses the net_pkt directly or copies the data if the data in the pkt is split to two buffers when called with `net_pkt_get_data()`